src/alligator/AlligatorOP_V5.sol

```diff
@@ -80,8 +80,14 @@ contract AlligatorOPV5 is IAlligatorOPV5, UUPSUpgradeable, OwnableUpgradeable, P
     // Subdelegation rules `from` => `to`
     mapping(address from => mapping(address to => SubdelegationRules subdelegationRules)) public subdelegations;
 
-    // Records of votes cast across an authority chain, to prevent double voting from the same proxy
-    mapping(address proxy => mapping(uint256 proposalId => mapping(address voter => uint256))) public votesCast;
+    mapping(address proxy => mapping(uint256 proposalId => mapping(address voter => uint256))) private UNUSED_SLOT;
+
+    // Records of votes cast on `proposalId` by `delegate` with `proxy` voting power from those subdelegated by `delegator`.
+    // Used to prevent double voting from the same proxy and authority chain.
+    mapping(
+        address proxy
+            => mapping(uint256 proposalId => mapping(address delegator => mapping(address delegate => uint256)))
+    ) public votesCast;
```

Comment:
this might not be enough to allow voting from the same proxy in some cases.

Consider the following chains:
`A -> B -> C -> D`
`A -> E -> C -> D`

when `D` votes, both the `proxy` and `delegator` will be the same which will prevent them from fully using their voting power

Reply:
this is by design as otherwise one could vote twice with same proxy -- when C gives an allowance to D, it applies to both authority chains. so if D votes through one, the votes used will be reduced when voting from the other.